test/multidevice.h

```diff
@@ -99,7 +99,11 @@ class PipelineTest : public MultiDeviceTest {
   std::vector<c10::IValue> inputs;
   std::vector<c10::IValue> unsharded_inputs;
   std::vector<at::Tensor> outputs;
+  std::vector<at::Tensor> ref_unsharded_outputs;
   MultiDeviceExecutorParams multi_device_executor_params;
+  // if true, the validation is performed with prescribed reference data stored
+  // in ref_unsharded_outputs. Otherwise, this data is computed automatically
+  bool validate_with_prescribed_values = false;
```

Comment:
How do I reason about these "global" (to the test) variables? They can be modified and read anywhere in the test. What are the common recipes?

Reply:
> The flag is set in the test body and read in the method validate(). 

That's the problem I want to avoid. Instead of communicating with `validate` via some global-to-class variables, I think it's better to pass the same information in via the parameters of `validate`. This makes information flow more explicit. I'll read the test more closely to see whether it's possible to do so.

Reply:
ok, I removed `validate_with_prescribed_values` as a class member and now pass this argument around the method